invenio_vocabularies/datastreams/readers.py

```diff
@@ -76,15 +95,87 @@ def __init__(
         self.content_type = content_type
         super().__init__(origin, *args, **kwargs)
 
-    def read(self):
-        """Reads a yaml file and returns a dictionary per element."""
+    def _read(self, fp, *args, **kwargs):
```

Comment:
The `fp` variable name loses a bit of its meaning here. In that case, maybe dropping it an using something like `url` is better (since we're overriding `read` as well),